crates/release_plz_core/src/repo_versions.rs

```diff
@@ -0,0 +1,50 @@
+use git_cmd::Repo;
+use regex::Regex;
+
+/// Returns the latest tag of the repository in the form of `vX.X.X`
+/// * `None` if there are no version tags found matching the regex,
+/// * `Tag` if at least one tag is matching the regex
+pub fn get_repo_versions(repo: &Repo) -> Option<String> {
```

Comment:
this function name is plural, but you are returning a string.

Also, workspaces tags are in the form of `{package}-v{version}`. E.g. https://github.com/MarcoIeni/release-plz/tags
Does the implementation of this PR recognize the right tag for the right package?

Reply:
> should we use the tag a-v1.0.0 as the comparison for the workspace repo, or should we checkout to b-v1.0.0?

I would use tag `a-v1.0.0` if we are comparing `a`, and tag `bv-1.0.0` if we are comparing `b`.
This mirrors better the current behavior of release-plz, when we compare each package with the version published in crates.io

> in tag b-v1.0.0, the version of package a is also v1.0.0, so nothing changed about a.

I'm not sure about this, maybe people released `b`, but they didn't release `a` yet. But this new tag contains unreleased changes of `a`.

The difference is calculated by analyzing every commit that touches the files of a package. Each commit is added to the changelog.
So I don't see issues of duplication across release notes. 🤔

Reply:
> maybe people released b, but they didn't release a yet.

Makes sense! I'll update the logic according to this, so it will detect the latest version for each package